packages/two_dimensional_scrollables/lib/src/table_view/table_delegate.dart

```diff
@@ -21,7 +21,7 @@ typedef TableSpanBuilder = TableSpan Function(int index);
 ///
 /// Used by [TableCellBuilderDelegate.builder] to build cells on demand for the
 /// table.
-typedef TableViewCellBuilder = Widget? Function(
+typedef TableViewCellBuilder = TableViewCell? Function(
```

Comment:
The verbosity it adds to uses cases that don't need to specify anything on the TableViewCell is slightly annoying (see some of the edits made to existing test cases). I don't have any better ideas on this one either right now, though.

Reply:
> For my own understanding: Is the only reason this is a TableViewCell instead of a basic Widget to ensure that we can auto-insert RepaintBoundaries around each cell? And we cannot have the delegate add the repaint boundaries because that would hide the TableViewCell-provided parent data?

Yes. I cannot use `is` in the delegate to conditionally apply the `RepaintBoundary`.

In the framework, in the non scrolling `Table`, we allow `TableCell` to be used or `Widget` to be used, and it very similarly applies `ParentData`, but here I could not figure another way around the `RepaintBoundary`. I feel like the performance is worth the minor annoyance, we may also add other features to `TableViewCell` in the future that would make it less singular in its use case.

Reply:
I think we can live with the extra verbosity. If we figure out a smarter way to handle the RepaintBoundary it should be non-breaking to widen the type here and allow any Widget.